line_profiler/_line_profiler.pyx

```diff
@@ -1,12 +1,50 @@
-from .python25 cimport PyFrameObject, PyObject, PyStringObject
+from python25 cimport PyFrameObject, PyObject, PyStringObject
+from sys import byteorder
+cimport cython
+from cpython.version cimport PY_VERSION_HEX
+from libc.stdint cimport int64_t
 
+from libcpp.unordered_map cimport unordered_map
+
+# long long int is at least 64 bytes assuming c99
+ctypedef unsigned long long int uint64
+ctypedef long long int int64
 
 cdef extern from "frameobject.h":
+    """
+    inline PyObject* get_frame_code(PyFrameObject* frame) {
```

Comment:
Can you explain this or send me a pointer to a reference on what's going on here? To me this looks like one version of the Python code is incrementing a reference count an never decrementing it, and on the other side I see a decrement with no corresponding increment. 

This is working around directly referencing `frame.f_code` in "python land"? It this just doing exactly what Python would do if you wrote `frame.f_code`? If not, what is it skipping, and if so wouldn't just using the attribute access be cleaner?

Reply:
By INCREFing, Python knows not to deallocate the co_code object until we destroy it. By DECREFing, we remove the reference that PyCode_GetCode grabs from the `code` object, since we're not using it, we're returning `ret`.

This is doing what the python would do, except it results in us not needed to put a runtime check for 3.11 or <3.11 because of the magic of C preprocessors and cython detecting which version of Python we're compiling for.

Reply:
I'm going to need to look into this a bit more. I want to completely understand this before merging. Because this is dealing with reference counting I want to ensure we aren't introducing a security issue.

Reply:
Sounds good. If you're familiar with memray, which is a cython memory profiler, this code is very similar to theirs: https://github.com/bloomberg/memray/blob/main/src/memray/_memray/compat.h#L38-L51

Reply:
I got this code directly from the person who made memray:
![Screenshot from 2022-09-23 12-29-49](https://user-images.githubusercontent.com/43764914/192008800-78f6ef1e-5344-4266-af7f-734c8eb7ca0f.png)